src/rocker/mount_extension.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2019 Open Source Robotics Foundation
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from rocker.extensions import RockerExtension
+
+
+class Mount(RockerExtension):
+
+    name = 'mount'
+
+    @classmethod
+    def get_name(cls):
+        return cls.name
+
+    def precondition_environment(self, cli_args):
+        pass
+
+    def validate_environment(self, cli_args):
+        pass
+
+    def get_preamble(self, cli_args):
+        return ''
+
+    def get_snippet(self, cli_args):
+        return ''
+
+    def get_docker_args(self, cli_args):
+        args = ['']
+        for mount in cli_args['mount']:
+            mount = os.path.abspath(mount)
+            args.append('-v {0}:{0}'.format(mount))
```

Comment:
Instead of forcing it to be in the same place on the target, could we just pass through the argument directly. Then we can rely on the default docker parsing of the volume argument.  Otherwise we're basically preventing the user from doing more expressive interactions. 
```suggestion
            args.append('-v {0}'.format(mount))
```

Reply:
It looks like the default parsing of the docker argument is a lot less convenient. I have a folder `~/ws/rocker`. Without this change I can run `rocker --mount ~/ws/rocker ubuntu:18.04 bash` and see the content in `~/ws/rocker`.

With this this change, the same command creates an empty folder `/home/sloretz/ws/rocker` inside the container. To get the same behavior as before I have to run `rocker --mount ~/ws/rocker:/home/sloretz/ws/rocker ubuntu:18.04 bash`.

Reply:
I suggested a solution in https://github.com/sloretz/rocker/pull/1.